### PR TITLE
WT-8988 Add a test for exclusive setting in create for tiered storage

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

**Comment by mm-ng** [comment]
I tried to refactor the code but I'm open to any other suggestions if there's a more intuitive way to do this.

File: test/suite/test_excl.py [hunk 1]
```diff
@@ -26,23 +26,79 @@
...
+
+    types = [
+        ('file', dict(type = 'file', uri_type = 'file:')),
+        ('table', dict(type = 'table', uri_type = 'table:')),
+        ('tiered', dict(type = 'tiered', uri_type = 'table:')),
+    ]
+
+    scenarios = make_scenarios(storage_sources, types)
+
+    def conn_config(self):
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_excl.py:54]: If we add `none_tiered` scenario we can just return an empty string here.

File: test/suite/test_excl.py [hunk 2]
```diff
@@ -26,23 +26,79 @@
...
+        if self.ss_name == 'dir_store' and not os.path.exists(self.bucket):
+            os.mkdir(self.bucket)
+        return \
+          'tiered_storage=(auth_token=%s,' % self.auth_token + \
+          'bucket=%s,' % self.bucket + \
+          'bucket_prefix=%s,' % self.bucket_prefix + \
+          'name=%s),tiered_manager=(wait=0)' % self.ss_name
+
+    def conn_extensions(self, extlist):
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_excl.py:63]: If we add `none_tiered` scenario we can just return an empty string here.

File: test/suite/test_excl.py [hunk 3]
```diff
@@ -26,23 +26,79 @@
...
+        ('s3', dict(auth_token = get_auth_token('s3_store'),
+            bucket = get_bucket1_name('s3_store'),
+            bucket_prefix = generate_s3_prefix(),
+            ss_name = 's3_store')),
+    ]
+
+    types = [
+        ('file', dict(type = 'file', uri_type = 'file:')),
+        ('table', dict(type = 'table', uri_type = 'table:')),
+        ('tiered', dict(type = 'tiered', uri_type = 'table:')),
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_excl.py:49]: Do we need 'tiered' here?

File: test/suite/test_excl.py [hunk 4]
```diff
@@ -26,23 +26,79 @@
...
-    # Create the object with "exclusive", then assert that creation with
-    # "exclusive" fails.
     def test_create_excl(self):
-        uri = self.type + 'create_excl'
-        self.session.create(uri, "exclusive")
+        uri = self.uri_type + "create_excl_" + self.type
+        exclusive_config = "exclusive=true"
+        not_exclusive_config = "exclusive=false"
+
+        # As we are using connection level tiered storage configurations specify that
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_excl.py:81]: If we add `none_tiered` scenario we don't need to control table type here and the `if` block below can be removed.

**Review by AlexeyAnisimovAU — COMMENTED** [review]
Good start! I have one suggestion and a few notes if we decided to implement it.

File: test/suite/test_excl.py [hunk 5]
```diff
@@ -26,23 +26,79 @@
...
-    ])
+    storage_sources = [
+        ('dir_store', dict(auth_token = get_auth_token('dir_store'),
+            bucket = get_bucket1_name('dir_store'),
+            bucket_prefix = "pfx_",
+            ss_name = 'dir_store')),
+        ('s3', dict(auth_token = get_auth_token('s3_store'),
+            bucket = get_bucket1_name('s3_store'),
+            bucket_prefix = generate_s3_prefix(),
+            ss_name = 's3_store')),
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply test/suite/test_excl.py:43]: This is a good suggestion.

**Review by sueloverso — APPROVED** [review]
LGTM - I agree with Alexey's suggestion to have a scenario that skips the tiered stuff.

File: test/suite/test_excl.py [hunk 6]
```diff
@@ -26,23 +26,40 @@
...
+        # Exclusive re-create should error.
         self.assertRaises(wiredtiger.WiredTigerError,
-            lambda: self.session.create(uri, "exclusive"))
+            lambda: self.session.create(uri, "exclusive=true"))
+
+        # Non-exclusive re-create is allowed.
+        self.session.create(uri, "exclusive=false")
+
+        # Exclusive create on a table that does not exist should succeed.
+        self.session.create(uri + "_non_existent", "exclusive=true")
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_excl.py:59]: Isn't the `uri + _non_existent` test the same as `uri` above? Maybe we can remove `_non_existent` test?

Comment (mm-ng) [reply test/suite/test_excl.py:59]: It's slightly different as these tables were not previously created and we're just testing the exclusive setting with the first create call. The one above test the exclusive setting after the table has already been created. But we can also remove these situations if they are not needed.

Comment (AlexeyAnisimovAU) [reply test/suite/test_excl.py:59]: All good, let's leave it there.

**Review by AlexeyAnisimovAU — APPROVED** [review]
LGTM!

**Pull request merged**